src/stores/GlobalStore.js

```diff
@@ -61,8 +84,23 @@ export default class GlobalStore {
     this.localWallet = Boolean(process.env.LOCAL_WALLET === 'true');
 
     // Call syncInfo once to init the wallet addresses used by other stores
-    this.getSyncInfo();
     this.subscribeSyncInfo();
+
+    wsLink.subscriptionClient.onConnected(this.setOnline, this);
+    wsLink.subscriptionClient.onReconnected(this.setOnline, this);
+    wsLink.subscriptionClient.onDisconnected(this.setOffline, this);
+    wsLink.subscriptionClient.onReconnecting(this.reconnecting, this);
+
+    // Subscribe to changes
+    window.addEventListener('offline', () => {
+      console.log('internet offline');
+      this.internetOnline = false;
+    });
+    window.addEventListener('online', () => {
```

Comment:
does this work? we were using `navigator.onLine` to check before. this logic is different.

Reply:
Yes, it works. Previously in `BottomBar`, we had both `navigator.onLine` and `window.listener`, but they don't affect each other. When the browser is online, it sets both `navigator.onLine` to true and triggers `window.listener` at the same time, so only one of them is enough.

In my case, I chose to use observable because this way, other components can react to it, also the `@computed get online` can react to both `socketOnline` and `internetOnline`.

Reply:
I am wondering why there both methods to check. I am thinking that maybe `navigator.onLine` handles some browsers and `window.addEventListener('online'` handles other browsers.

Reply:
I tested on Safari and Firefox, both work fine.

The original uses listeners to set or unset a variable called `online` while this variable is never used.